tests/unit/test_util.py

```diff
@@ -15,186 +13,175 @@
+    required_distribution_deps = get_required_distribution_deps("pybamm")
 
-    def test_get_parameters_filepath(self):
-        tempfile_obj = tempfile.NamedTemporaryFile("w", dir=".")
-        self.assertTrue(
-            pybamm.get_parameters_filepath(tempfile_obj.name) == tempfile_obj.name
-        )
-        tempfile_obj.close()
-
-        package_dir = os.path.join(pybamm.root_dir(), "pybamm")
-        tempfile_obj = tempfile.NamedTemporaryFile("w", dir=package_dir)
-        path = os.path.join(package_dir, tempfile_obj.name)
-        self.assertTrue(pybamm.get_parameters_filepath(tempfile_obj.name) == path)
-        tempfile_obj.close()
-
-    def test_is_jax_compatible(self):
-        if pybamm.have_jax():
-            compatible = pybamm.is_jax_compatible()
-            self.assertTrue(compatible)
-
-    def test_git_commit_info(self):
-        git_commit_info = pybamm.get_git_commit_info()
-        self.assertIsInstance(git_commit_info, str)
-        self.assertEqual(git_commit_info[:2], "v2")
-
-    def test_import_optional_dependency(self):
-        optional_distribution_deps = get_optional_distribution_deps("pybamm")
-        present_optional_import_deps = get_present_optional_import_deps(
-            "pybamm", optional_distribution_deps=optional_distribution_deps
+    # Get nested required dependencies
+    for distribution_dep in list(required_distribution_deps):
+        required_distribution_deps.update(
+            get_required_distribution_deps(distribution_dep)
         )
 
-        # Save optional dependencies, then make them not importable
-        modules = {}
-        for import_pkg in present_optional_import_deps:
-            modules[import_pkg] = sys.modules.get(import_pkg)
-            sys.modules[import_pkg] = None
-
-        # Test import optional dependency
-        for import_pkg in present_optional_import_deps:
-            with self.assertRaisesRegex(
-                ModuleNotFoundError,
-                f"Optional dependency {import_pkg} is not available.",
-            ):
-                pybamm.util.import_optional_dependency(import_pkg)
-
-        # Restore optional dependencies
-        for import_pkg in present_optional_import_deps:
-            sys.modules[import_pkg] = modules[import_pkg]
-
-    def test_pybamm_import(self):
-        optional_distribution_deps = get_optional_distribution_deps("pybamm")
-        present_optional_import_deps = get_present_optional_import_deps(
-            "pybamm", optional_distribution_deps=optional_distribution_deps
+    # Check that optional dependencies are not present in the core PyBaMM installation
+    optional_present_deps = optional_distribution_deps & required_distribution_deps
+    assert not bool(optional_present_deps), (
+        f"Optional dependencies installed: {optional_present_deps}.\n"
+        "Please ensure that optional dependencies are not present in the core PyBaMM installation, "
+        "or list them as required."
+    )
```

Comment:
I don't think this is correct. While it may have been passing, `&` is used for bitwise operations, the `and` keyword is better to use here, i.e., for boolean operations. Also, you should cast `optional_present_deps` as boolean prior to the assert statement.

Reply:
I think `optional_distribution_deps` and `required_distribution_deps` here are sets and hence `&` checks for the common elements if found. Using `and` here leads to test failure.

Reply:
Ah, I see, I think we should fix that in a different PR. Thanks for looking!